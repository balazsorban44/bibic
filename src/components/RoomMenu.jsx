import React from 'react'
import {withStore} from "./db"
import {Link} from 'react-scroll'


export const RoomMenu = ({
  rooms, isBigScreen, onClick
}) =>
  rooms.map(({id}) =>
    <li key={id}>
      <Link
        offset={isBigScreen ? -106 : -40}
        onClick={onClick}
        to={`szoba-${id}`}
      >
        {id}
      </Link>
    </li>
  )

export default withStore(RoomMenu)