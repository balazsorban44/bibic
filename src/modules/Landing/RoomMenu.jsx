import React from "react"
import {Link} from "react-scroll"

import useRoom from "hooks/data/useRoom"
import Loading from "components/Loading"

export default function RoomMenu ({mdUp, onClick, open}) {

  const [rooms, loading] = useRoom()

  return (
    <ul className={`room-menu ${open ? "room-menu-show" : ""}`}>
      {loading ? <Loading/> : rooms.map(RoomMenuItem({mdUp, onClick}))}
    </ul>
  )
}

const RoomMenuItem = ({mdUp, ...props}) => ({id}) =>
  <li key={id}>
    <Link
      {...props}
      offset={mdUp ? -106 : -40}
      to={`szoba-${id}`}
    >
      {id}
    </Link>
  </li>