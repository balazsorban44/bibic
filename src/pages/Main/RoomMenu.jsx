import React from 'react'
import {useStore} from "../../context/db"
import {Link} from 'react-scroll'


export const RoomMenu = ({isBigScreen, onClick}) => {
  const {rooms} = useStore()
  return rooms.map(({id}) =>
    <li key={id}>
      <Link
        offset={isBigScreen ? -106 : -40}
        onClick={onClick}
        to={`szoba-${id}`}
      >
        {id}
      </Link>
    </li>
  )
}

export default RoomMenu