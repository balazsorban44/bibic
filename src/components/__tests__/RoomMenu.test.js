import {RoomMenu} from "../RoomMenu"
import {Link} from 'react-scroll'

describe("RoomMenu component", () => {
  const props = {
    rooms: [{id: 1}], isBigScreen: true, onClick: jest.fn()
  }
  const wrapper = shallow(<RoomMenu {...props}/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe("different screen size", () => {
    it("big screen", () => {
      expect(wrapper.find(Link).first().prop("offset")).toBe(-106)
    })
    it("small screen", () => {
      wrapper.setProps({isBigScreen: false})
      expect(wrapper.find(Link).first().prop("offset")).toBe(-40)
    })

  })

  it("renders li element for each room", () => {
    expect(wrapper.find("li").length).toBe(props.rooms.length)
  })
})