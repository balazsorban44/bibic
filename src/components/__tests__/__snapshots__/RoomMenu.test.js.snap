// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomMenu component renders correctly 1`] = `ShallowWrapper {}`;
